app/controllers/user.js

```diff
@@ -20,6 +23,26 @@ exports.register = (req, res, next) => {
     .catch(next);
 };
 
+exports.registerAdmin = (req, res, next) => {
+  const user = createUserObject(req);
+  user.role = admin_role;
+  userService
+    .findOne({ email: user.email })
+    .then(usr => {
+      if (usr) {
+        usr.role = admin_role;
+        userService.updateUserRole(usr).then(updatedUser => {
+          res.send(200, updatedUser);
+        });
+      } else {
+        userService.createUser(user).then(createdUser => {
```

Comment:
Add return in promises, also check more simple way of update or insert some model